hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala

```diff
@@ -121,8 +122,10 @@ object TableMapIRNew {
         is.seek(filePosition)
         val aggSize = is.readInt()
         val partAggs = new Array[Byte](aggSize)
-        val nread = is.read(partAggs)
-        assert(nread == aggSize)
+        val nread = is.read(partAggs, 0, aggSize)
```

Comment:
It seems like this read method is not conforming to expected behavior for Java.

```
is.hail.utils.HailException: aggs read wrong number of bytes: 65232 vs 2820547
```
I tried updating this pr with, and it seems to be working?
```diff
diff --git a/hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala b/hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala
index 9664a03b6..8695dd16a 100644
--- a/hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala
+++ b/hail/src/main/scala/is/hail/expr/ir/agg/Extract.scala
@@ -122,7 +122,12 @@ object TableMapIRNew {
         is.seek(filePosition)
         val aggSize = is.readInt()
         val partAggs = new Array[Byte](aggSize)
-        val nread = is.read(partAggs, 0, aggSize)
+        var nread = is.read(partAggs, 0, aggSize)
+        var r = nread
+        while (r > 0 && nread < aggSize) {
+          r = is.read(partAggs, nread, aggSize - nread)
+          if (r > 0) nread += r
+        }
         if (nread != aggSize) {
           fatal(s"aggs read wrong number of bytes: $nread vs $aggSize")
         }
```